src/ripple_app/ledger/LedgerEntrySet.cpp

```diff
@@ -1122,6 +1122,23 @@ STAmount LedgerEntrySet::rippleOwed (const uint160& uToAccountID, const uint160&
 
     if (sleRippleState)
     {
+        { // check freeze
+            
+            SLE::pointer sleIssuer = entryCache(ltACCOUNT_ROOT,
+                Ledger::getAccountRootIndex(uToAccountID));
+            if(sleIssuer)
+            {
+                if(sleIssuer->isFlag(lsfRequireAuth))
+                {
+                    if(sleRippleState->isFlag(
+                        (uFromAccountID > uToAccountID) ? lsfHighAuth : lsfLowAuth))
```

Comment:
I am not sure I follow why this is helpful in enforcing a freeze:
you're still returning saBalance (but with the wrong sign in half the instance)